### PR TITLE
Use the new Pennylane Sphinx Theme

### PR DESCRIPTION
**Pull request by josh146**
**Context:** Replaces the Xanadu Sphinx Theme with the PennyLane Sphinx Theme, which centralizes colours, navbar, and footer information.

**Description of the Change:** As above.

**Benefits:** Easier to update non-lightning specific information (like navbar links) without having to push to Lightning.

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

**Comment by josh146** [comment]
@mlxd I just realized - lightning uses `stable` as the default docs build? This may require a 'postfix' release in order for the docs to update after merging.

**Review by mlxd — APPROVED** [review]
Nice work again!

File: doc/conf.py [hunk 1]
```diff
@@ -199,86 +200,19 @@ def __getattr__(cls, name):
...
-# Custom sidebar templates, must be a dictionary that maps document names
-# to template names.
-html_sidebars = {
-    "**": [
-        "searchbox.html",
-        "globaltoc.html",
-    ]
-}
-
 # -- Xanadu theme ---------------------------------------------------------
```
Comment (AmintorDusko) [inline doc/conf.py:203]:
```suggestion
# -- html theme ---------------------------------------------------------
```

File: doc/conf.py [hunk 2]
```diff
@@ -199,86 +200,19 @@ def __getattr__(cls, name):
...
-html_theme = "xanadu"
+html_theme = "pennylane"
 
 # xanadu theme options (see theme.conf for more information)
```
Comment (AmintorDusko) [inline doc/conf.py:206]:
```suggestion
# html theme options (see theme.conf for more information)
```

**Review by AmintorDusko — APPROVED** [review]
Just a minor suggestion.
Beautiful : 
image

Happy to approve!

**Pull request merged**